gateway/src/main/scala/com/improving/app/gateway/infrastructure/routes/EventGatewayRoutes.scala

```diff
@@ -45,23 +52,92 @@ trait EventGatewayRoutes extends ErrorAccumulatingCirceSupport with StrictLoggin
                     fromJsonString[GatewayCancelEvent](data)
                   )
               ) { eventCancelled =>
-                complete(JsonFormat.toJsonString(eventCancelled))
+                complete(eventCancelled.print)
               }
             }
           }
-        } ~
-          post {
-            entity(Directives.as[String]) { data =>
-              onSuccess(
-                handler
-                  .createEvent(
-                    fromJsonString[GatewayCreateEvent](data)
+        } ~ pathPrefix("allIds") {
+          get {
+            onSuccess(
+              handler.getAllIds
+            ) { allIds =>
+              complete(allIds.print)
+            }
+          }
+        } ~ pathPrefix("allData") {
+          pathPrefix("status") {
+            pathPrefix(Segment) { status =>
+              pathPrefix("forOrg") {
+                pathPrefix(Segment) { orgId =>
+                  get {
+                    onSuccess(
+                      handler.getAllIds.map(_.allEventIds.map(id => handler.getEventData(id.id)))
+                    ) { allIdsFut =>
+                      complete(
+                        Future
+                          .sequence(allIdsFut)
+                          .map(data =>
+                            data
+                              .filter(_.eventMetaInfo.map(_.currentState) == EventState.fromName(status))
+                              .filter(
+                                _.eventInfo.exists(info =>
+                                  if (info.value.isInfo) info.getInfo.sponsoringOrg.contains(OrganizationId(orgId))
+                                  else info.getEditable.sponsoringOrg.contains(OrganizationId(orgId))
+                                )
+                              )
+                              .map(_.print)
+                          )
+                      )
+                    }
+                  }
+                }
+              } ~ get {
+                onSuccess(
+                  handler.getAllIds.map(_.allEventIds.map(id => handler.getEventData(id.id)))
+                ) { allIdsFut =>
+                  complete(
+                    Future
+                      .sequence(allIdsFut)
+                      .map(data =>
+                        data
+                          .filter(_.eventMetaInfo.map(_.currentState) == EventState.fromName(status))
+                          .map(_.print)
+                      )
                   )
-              ) { eventCreated =>
-                complete(JsonFormat.toJsonString(eventCreated))
+                }
               }
             }
+          } ~ get {
+            onSuccess(
+              handler.getAllIds.map(_.allEventIds.map(id => handler.getEventData(id.id)))
```

Comment:
This looks like slightly more logic than I'd want in the routes, and it's all handler stuff, so maybe a handler method?